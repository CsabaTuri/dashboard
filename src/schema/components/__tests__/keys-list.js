import React from "react"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import { query as q } from "faunadb"

import KeysList from "../keys-list"
import { reduceSchemaTree } from "../../"

jest.mock("react-router", () => ({
  browserHistory: {
    push: jest.fn()
  }
}))

const { browserHistory } = require("react-router")

describe("KeysList Component", () => {
  let comp, client, store

  beforeEach(() => {
    client = {
      query: jest.fn(() =>
        Promise.resolve("fake-key")
      )
    }

    store = createImmutableTestStore({ schema: reduceSchemaTree })()
    comp = shallow(<KeysList store={store} client={client} />).dive()
  })

  it("should render an empty page", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("fetches keys", () => {
    comp.find("Connect(Pagination)").prop("query")()
    expect(client.query).toHaveBeenCalled()
  })

  it("fetch a key when clicked", () => {
    comp.find("Connect(Pagination)").simulate("selectRef", q.Ref("keys/123"))
    expect(client.query).toHaveBeenCalled()
  })

  it("displays the selected key", () => {
    return comp.instance().onRefSelected(q.Ref("keys/123")).then(() => {
      comp.update()
      expect(shallowToJson(comp)).toMatchSnapshot() // returned key will be present
    })
  })

  it("redirect to database when database ref is cliekced", () => {
    comp.instance().onRefSelected(q.Ref("databases/my-blog"))
    expect(browserHistory.push).toHaveBeenCalledWith("/db/my-blog/databases")
  })
})
