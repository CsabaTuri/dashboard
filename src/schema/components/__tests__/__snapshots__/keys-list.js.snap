exports[`KeysList Component should render an empty page 1`] = `
<InstancesList
  onSelectRef={[Function]}
  query={[Function]} />
`;
